src/Nethermind/Nethermind.Evm/EOF/EvmObjectFormat.cs

```diff
@@ -357,13 +360,17 @@ bool ValidateInstructions(ReadOnlySpan<byte> code, in EofHeader header)
                     }
 
                     var immediateValueSize = ONE_BYTE_LENGTH + count * TWO_BYTE_LENGTH;
-                    immediates.Add(new Range(pos, pos + immediateValueSize - 1));
+
+                    for (int j = pos; j <= pos + immediateValueSize - 1; j++)
+                    {
+                        immediatesMask.Set(j, true);
+                    }
```

Comment:
You missed another point of the speedup, the point is you can now set multiple bits at once, instead of doing a loop.

Reply:
would this be a better alternative ? 
```csharp
public static void SetBits(this BitArray thisArr, bool value, Range range)
{
    long rangeMasked = (1 << range.End.Value) - (1 << range.Start.Value);
    BitArray arr = new BitArray(rangeMasked.ToBigEndianByteArray());
    thisArr.Or(arr);
}
```
or is there a better way to do it ?

as for benchmarks , I will try to generate them asap

Reply:
Yes for low number of bits it is best to use masks like in `CodeDataAnalyzer`